wazuh/wazuh
# Enhance CA handling and error messaging for improved configuration usability

## Objective
This PR enhances the system's ability to handle multiple Certificate Authorities (CAs) by modifying the logic to read multiple CA certificates from the configuration and concatenate them into a single file. Additionally, we have improved error messaging to enhance usability, aligning with the Ghezzi quality software characteristics.

## Description
We implemented a new logic that allows the system to read multiple CA entries from the configuration and concatenate them into a single CA bundle file. This streamlined approach simplifies the setup process for scenarios requiring multiple CAs, this approach has been validated as 

-  #24551

We also recognized the need to improve the clarity and helpfulness of error messages related to CA configuration. Following the Ghezzi quality software characteristic of usability, we refined our error messaging to be more descriptive, guiding users more effectively through resolving configuration issues.

## Quality Assurance

### Static Analysis
N/A

### Testing

#### Unit Testing
N/A

#### Integration Testing
- Integration testing was conducted to ensure that the system correctly handles the new CA bundle during runtime, especially in environments requiring secure communications authenticated by multiple CAs.
- **Testing Steps**:
  1. Configure the system with multiple valid and invalid CA entries. Follow the first step of #24551
  2. Verify that the system concatenates valid CAs correctly and utilizes the resulting CA bundle for secure connections. Use the `cat` command for example and `stat` to check file permissions.
  3. Run manager, and indexer connection should be successful.

<details><summary>Screenshots</summary>

![image](https://github.com/user-attachments/assets/4b95900e-6ec1-4238-82c1-7f6184abd4ca)

![Pasted image](https://github.com/user-attachments/assets/168ca2f8-092a-4360-b362-037fc636c338)

![image](https://github.com/user-attachments/assets/30e6382c-6eb3-4dc4-99ee-defcad559d0a)

![image](https://github.com/user-attachments/assets/baf00d60-a8a0-4d98-927b-692c42f064db)


</details>

## Additional Notes
- The path proposal can change according with reviewers.
